src/util/dump.ts

```diff
@@ -90,24 +91,16 @@ export async function downloadAndSaveModule(zimCreator: ZimCreator, mw: MediaWik
     // return :
     //   a promise resolving 1 if data has been succesfully saved or resolving 0 if data was already in Redis
 
-    // the 2 variable functions below are a hack to call startUp() (from module startup) when the 3 generic dependencies (startup, jquery, mediawiki) are loaded.
-    // on wikipedia, startUp() is called in the callback of the call to load.php to dl jquery and mediawiki but since load.php cannot be called in offline,
-    // this hack calls startUp() when custom event fireStartUp is received. Which is dispatched when module mediawiki has finished loading
+    // the function hackStartupModule changes startup script by returning true for all modules so that load.php is not called.
+    // it also removes requestIdleCallback as in our case window is idle after all script tags are called but those script tags
+    // will require the functions which would have been loaded by doPropagation.
     function hackStartUpModule(jsCode: string) {
-        return jsCode.replace(
-            'script=document.createElement(\'script\');',
-            `
-                    document.body.addEventListener('fireStartUp', function () { startUp() }, false);
-                    return;
-                    script=document.createElement('script');`,
-        );
-    }
```

Comment:
Why we remove this? It is not needed anymore?

Reply:
no it isn't there is actually no function `startup()` present anywhere in resource modules.